core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -28,15 +36,33 @@ import org.apache.spark.scheduler._
  *
  * This class is thread-safe (unlike JobProgressListener)
  */
+
+object StorageStatusListener {
+  val TIME_TO_EXPIRE_KILLED_EXECUTOR = "spark.ui.timeToExpireKilledExecutor"
+}
+
 @DeveloperApi
-class StorageStatusListener extends SparkListener {
+class StorageStatusListener private[storage](conf: SparkConf, ticker: Ticker) {
+  def this(conf: SparkConf) = {
+    this(conf, Ticker.systemTicker())
+  }
+  
+  import StorageStatusListener._
+  
   // This maintains only blocks that are cached (i.e. storage level is not StorageLevel.NONE)
   private[storage] val executorIdToStorageStatus = mutable.Map[String, StorageStatus]()
+  private[storage] val removedExecutorIdToStorageStatus = CacheBuilder.newBuilder().
+    expireAfterWrite(conf.getTimeAsSeconds(TIME_TO_EXPIRE_KILLED_EXECUTOR, "0"), TimeUnit.SECONDS).
+    ticker(ticker).build[String, StorageStatus]()
```

Comment:
super nit:
can you put each chained call on its own line, with the `.` at the beginning of the line? so

```scala
private[storage] val removedExecutorIdToStorageStatus = CacheBuilder.newBuilder()
  .expireAfterWrite(...)
  .ticker(ticker)
  .build...
```